packages/advanced-logic/src/extensions/payment-network/address-based.ts

```diff
@@ -1,292 +1,279 @@
+  TCreationParameters extends ExtensionTypes.PnAddressBased.ICreationParameters = ExtensionTypes.PnAddressBased.ICreationParameters
+> extends AbstractExtension<TCreationParameters> {
+  public constructor(
+    public extensionId: ExtensionTypes.ID,
+    public currentVersion: string,
+    public supportedNetworks: string[],
+    public supportedCurrencyType: string,
+  ) {
+    super(ExtensionTypes.TYPE.PAYMENT_NETWORK, extensionId, currentVersion);
+    this.actions = {
+      ...this.actions,
+      [ExtensionTypes.PnAddressBased.ACTION.ADD_PAYMENT_ADDRESS]: this.applyAddPaymentAddress.bind(
+        this,
+      ),
+      [ExtensionTypes.PnAddressBased.ACTION.ADD_REFUND_ADDRESS]: this.applyAddRefundAddress.bind(
+        this,
+      ),
+    };
+  }
 
-const CURRENT_VERSION = '0.1.0';
+  /**
+   * Creates the extensionsData for address based payment networks
+   *
+   * @param extensions extensions parameters to create
+   *
+   * @returns IExtensionCreationAction the extensionsData to be stored in the request
+   */
+  public createCreationAction(
+    creationParameters: TCreationParameters,
+  ): ExtensionTypes.IAction<TCreationParameters> {
+    if (
+      creationParameters.paymentAddress &&
+      !this.isValidAddress(creationParameters.paymentAddress)
+    ) {
+      throw Error('paymentAddress is not a valid address');
+    }
 
-/**
- * Creates the extensionsData for address based payment networks
- *
- * @param extensions extensions parameters to create
- *
- * @returns IExtensionCreationAction the extensionsData to be stored in the request
- */
-function createCreationAction(
-  extensionId: ExtensionTypes.ID,
-  creationParameters: ExtensionTypes.PnAddressBased.ICreationParameters,
-): ExtensionTypes.IAction {
-  return {
-    action: ExtensionTypes.PnAddressBased.ACTION.CREATE,
-    id: extensionId,
-    parameters: {
-      paymentAddress: creationParameters.paymentAddress,
-      refundAddress: creationParameters.refundAddress,
-    },
-    version: CURRENT_VERSION,
-  };
-}
+    if (
+      creationParameters.refundAddress &&
+      !this.isValidAddress(creationParameters.refundAddress)
+    ) {
+      throw Error('refundAddress is not a valid address');
+    }
 
-/**
- * Creates the extensionsData to add a payment address
- *
- * @param extensions extensions parameters to create
- *
- * @returns IAction the extensionsData to be stored in the request
- */
-function createAddPaymentAddressAction(
-  extensionId: ExtensionTypes.ID,
-  addPaymentAddressParameters: ExtensionTypes.PnAddressBased.IAddPaymentAddressParameters,
-): ExtensionTypes.IAction {
-  return {
-    action: ExtensionTypes.PnAddressBased.ACTION.ADD_PAYMENT_ADDRESS,
-    id: extensionId,
-    parameters: {
-      paymentAddress: addPaymentAddressParameters.paymentAddress,
-    },
-  };
-}
+    return super.createCreationAction(
+      creationParameters,
+    ) as ExtensionTypes.IAction<TCreationParameters>;
```

Comment:
```suggestion
    )
```
Maybe I'm missing something but I think it's not needed